web/federate.go

```diff
@@ -74,6 +74,16 @@ func (h *Handler) federation(w http.ResponseWriter, req *http.Request) {
 	}
 	sort.Sort(byName(vector))
 
+	externalLabels := h.externalLabels.Clone()
+	if _, ok := externalLabels[model.InstanceLabel]; !ok {
+		externalLabels[model.InstanceLabel] = ""
+	}
+	externalLabelnames := make([]string, 0, len(externalLabels))
```

Comment:
If you use `model.LabelNames` instead of `string`, you don't have to type-convert in L83.

`model.LabelNames` is also sortable, so you can write `sort.Sort(externalLabelnames)` in L85.